moveit_ros/hybrid_planning/local_planner/local_planner_component/include/moveit/local_planner/local_planner_component.h

```diff
@@ -118,6 +118,10 @@ class LocalPlannerComponent
       declareOrGetParam<std::string>("local_solution_topic_type", local_solution_topic_type, undefined, node);
       declareOrGetParam<bool>("publish_joint_positions", publish_joint_positions, false, node);
       declareOrGetParam<bool>("publish_joint_velocities", publish_joint_velocities, false, node);
+      // Planning scene monitor options
+      declareOrGetParam<std::string>("monitored_planning_scene", monitored_planning_scene_topic, undefined, node);
+      declareOrGetParam<std::string>("collision_object_topic", collision_object_topic, undefined, node);
+      declareOrGetParam<std::string>("joint_states_topic", joint_states_topic, undefined, node);
```

Comment:
Same question here about defaults.  Should the defaults for all these be defined somewhere (maybe a header file with a struct that has default values).

Reply:
MoveIt already does have that header file. I don't think we should use it, though -- I think we should enforce correct yaml.

(Using defaults leads to issues. See Henning's previous comment about using MoveItCppOptions and my response.)

Reply:
You don't have to provide a default when you call declare_parameter.

Reply:
I tried something like this. It seems nice for several reasons but it crashes.

```
void DeclareParam(const std::string& param_name, const rclcpp::Node::SharedPtr& node)
{
  try
  {
    if (node->has_parameter(param_name))
    {
      rcl_interfaces::msg::ParameterDescriptor descriptor;
      descriptor.dynamic_typing = true;
      node->declare_parameter(param_name, rclcpp::ParameterValue{}, descriptor);
    }
    else
      RCLCPP_ERROR_STREAM(node->get_logger(),
                          "Error getting parameter '" << param_name << "', check parameter type in YAML file");
  }
  catch (const rclcpp::exceptions::InvalidParameterTypeException& e)
  {
    RCLCPP_ERROR_STREAM(node->get_logger(),
                        "Error getting parameter '" << param_name << "', check parameter type in YAML file");
    throw e;
  }
}

std::string param_name = PLANNING_PIPELINES_NS + "pipeline_names";
DeclareParam(param_name, node);  --> Crashes because no default value is provided, I suppose
```

Reply:
has_parameter returns true if you have already declared the parameter, if you then call declare_parameter then it will throw an exception because it is already declared.

Reply:
That's a good point, maybe it will work without that. I'll try again.